basesandbox/src/ipc/semaphore.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+use nix::errno::Errno;
+use nix::libc;
+
+pub enum Mode {
+    CREATE,
+    OPEN,
+}
+
+/// Inter-process semaphore using POSIX
+pub struct Semaphore {
```

Comment:
Do we need to implement a semaphore? Isn't there any library for it?

Reply:
I couldn't find any. Any suggestion?